mev_inspect/aave_liquidations.py

```diff
@@ -63,26 +63,28 @@ def get_aave_liquidations(
 
 
 def _get_received_data(
-    child_traces: List[ClassifiedTrace], liquidator: str
+    liquidation: DecodedCallTrace, child_traces: List[ClassifiedTrace], liquidator: str
 ) -> Tuple[str, int]:
 
     """Look for and return liquidator payback from liquidation"""
-
     for child in child_traces:
 
         child_transfer: Optional[Transfer] = get_transfer(child)
 
-        if child_transfer is not None:
-
-            if child_transfer.to_address == liquidator:
+        if child_transfer is not None and child_transfer.to_address == liquidator:
+            return child_transfer.token_address, child_transfer.amount
 
-                return child_transfer.token_address, child_transfer.amount
+    if liquidation.error is not None:
+        return liquidation.inputs["_collateral"], 0
 
-    raise RuntimeError("Transfer from AAVE to liquidator not found!")
+    else:
+        raise RuntimeError(
+            f"No payback or input data found for liquidation in tx: {liquidation.transaction_hash}"
+        )
 
 
 def _get_debt_data(
```

Comment:
where does the PR address this part: "AaveCore contract calls a liquidation which is internally settled and does not result in any transfers" - that would still raise an exception after this change right?

Reply:
No, this would keep the purchase amount from the inputs and have no received=0

Reply:
in theory they "received" some though right?

for profit calculation should we just then filter out cases where received is 0?

Reply:
filter out as in returning None?

Reply:
like when we say "how much MEV $ came from liquidations" what should we do with aave liquidations